File: 415-add-strings/add-strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        int p1 = num1.length() - 1;
        int p2 = num2.length() - 1;
        int digit = 0;
        StringBuilder sb = new StringBuilder();
        while (p1 >=0 || p2 >=0 || digit > 0) {
            int n1 = 0, n2 = 0;
            if (p1 >= 0) {
                n1 = num1.charAt(p1) - '0';
                p1--;
            }
            if (p2 >= 0) {
                n2 = num2.charAt(p2) - '0';
                p2--;
            }
            int num = (n1 + n2 + digit) % 10;
            digit = (n1 + n2 + digit) / 10;
            sb.append(num);
        }
        return sb.reverse().toString();
    }
}

File: 415-add-strings/add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        n1 = list(num1)
        n2 = list(num2)
        res = []
        digits = 0
        i = 1
        while i <= (max(len(n1), len(n2))):
            if i > len(n1):
                a = 0
            else:
                a = int(n1[-i])
            if i > len(n2):
                b = 0
            else:
                b = int(n2[-i])
            # print(a,b)
            add = (a + b + digits) % 10
            digits = (a + b + digits) // 10
            res = [add] + res
            # print(res)
            i += 1
        if digits != 0:
            res = [digits] + res
            i += 1
        ret = ""
        for i in res:
            ret = ret + str(i)
        return ret